### PR TITLE
feat(slice-6): statement storage lifecycle — list + purge UI

### LINKED ISSUE #7: Slice 6: Statement storage lifecycle
## Parent

#1

## What to build

Finish the statement storage lifecycle so uploaded raw files are managed as first-class product artifacts rather than only backend implementation details.

This slice should cover the full user-facing lifecycle:
- persist raw files locally with stable metadata tied to statements;
- expose statement history/details so a user can inspect what was uploaded and when;
- detect identical re-uploads at the statement/file level using content hashing;
- add delete/purge flows with clear UX around what is reversible versus permanent;
- ensure file cleanup semantics are reliable and visible when purge succeeds or fails.

Implementation notes:
- the current backend already has part of the storage and purge path; extend it into a complete product slice rather than rebuilding it;
- be explicit about whether delete is soft-delete, purge is permanent, or both are supported with different actions.

## Acceptance criteria

- [ ] Raw uploaded files are stored under the configured local storage root with durable metadata in the database.
- [ ] Statement-level hash idempotency prevents duplicate statement artifacts from identical file uploads.
- [ ] The product shows a statement history/detail view with filename, upload time, size, and ingest outcome.
- [ ] Users can invoke the supported delete/purge actions from the UI and see the result clearly.
- [ ] Permanent purge removes the expected database records and underlying raw file bytes.
- [ ] Tests cover storage layout, idempotent re-upload behavior, and delete/purge edge cases.

## Blocked by

- #5
- #6

### PR DESCRIPTION
**Pull request by chandra-jagarlamudi**
Implements Slice 6: adds GET /statements and GET /statements/{id} endpoints, statement list page with purge confirmation, wired into the app shell nav.

Closes #7

### CODE REVIEW

File: frontend/src/StatementsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  const [loading, setLoading] = useState(true)
+  const [banner, setBanner] = useState<Banner | null>(null)
+  const [confirmId, setConfirmId] = useState<string | null>(null)
+  const [purging, setPurging] = useState(false)
+
+  async function load() {
+    setLoading(true)
+    try {
+      const data = await listStatements()
+      setStatements(data)
```
[the remark this answers is not in the document]
Comment (chandra-jagarlamudi) [reply frontend/src/StatementsPage.tsx:29]: @copilot apply changes based on [this feedback](https://github.com/chandra-jagarlamudi/PersonalFinancialAnalyst/pull/33#discussion_r3175980241)

File: frontend/src/StatementsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  if (bytes < 1024) return `${bytes} B`
+  if (bytes < 1024 * 1024) return `${(bytes / 1024).toFixed(1)} KB`
+  return `${(bytes / (1024 * 1024)).toFixed(1)} MB`
+}
+
+function formatDate(iso: string): string {
+  return new Date(iso).toLocaleString()
+}
+
+export default function StatementsPage() {
```
[the remark this answers is not in the document]
Comment (chandra-jagarlamudi) [reply frontend/src/StatementsPage.tsx:16]: @copilot apply changes based on [this feedback](https://github.com/chandra-jagarlamudi/PersonalFinancialAnalyst/pull/33#discussion_r3175980256)

File: frontend/src/StatementsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+              <th>Size</th>
+              <th>Inserted</th>
+              <th>Skipped</th>
+              <th></th>
+            </tr>
+          </thead>
+          <tbody>
+            {statements.map(s => (
+              <tr key={s.id}>
+                <td>{accounts.get(s.account_id)?.name ?? s.account_id}</td>
```
[the remark this answers is not in the document]
Comment (chandra-jagarlamudi) [reply frontend/src/StatementsPage.tsx]: @copilot apply changes based on [this feedback](https://github.com/chandra-jagarlamudi/PersonalFinancialAnalyst/pull/33#discussion_r3176006742)

File: frontend/src/StatementsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  if (bytes < 1024) return `${bytes} B`
+  if (bytes < 1024 * 1024) return `${(bytes / 1024).toFixed(1)} KB`
+  return `${(bytes / (1024 * 1024)).toFixed(1)} MB`
+}
+
+function formatDate(iso: string): string {
+  return new Date(iso).toLocaleString()
+}
+
+export default function StatementsPage() {
```
[the remark this answers is not in the document]
Comment (chandra-jagarlamudi) [reply frontend/src/StatementsPage.tsx:16]: @copilot apply changes based on [this feedback](https://github.com/chandra-jagarlamudi/PersonalFinancialAnalyst/pull/33#discussion_r3176006752)

File: backend/pfa/statements_api.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+class StatementOut(BaseModel):
+    id: UUID
+    account_id: UUID
+    filename: str
+    sha256: str
+    byte_size: int
+    inserted: int
+    skipped_duplicates: int
+    created_at: str
```
[the remark this answers is not in the document]
Comment (chandra-jagarlamudi) [reply backend/pfa/statements_api.py]: @copilot apply changes based on [this feedback](https://github.com/chandra-jagarlamudi/PersonalFinancialAnalyst/pull/33#discussion_r3176006757)
